### ISSUE TITLE
QOL suggestions

### ISSUE THREAD

**Issue by bpadovese** [coastal-science/HALLO-annotation#5]
I have some quality of life suggestions. I will be updating this issue if I have more.

- Tooltip is spectrogram settings. It wasnt immediately obvious what some settings were for some users. For instance, they though window length meant the length of the segment and not the stft setting for the spectrogram. I think a tooltip that users can hover over in some settings would be useful
- When you log in as an annotator, the system seems to fetch every segment available during login. This can take a while if there are many segments (5000+), in addition, it seemed like the app got extremely laggy afterwards (unusable really). I am not 100% sure this is what happened. Would it not be best to have a buffer of about 40 segments in the immediate vicinity of the one the user is annotating, instead of loading everything at startup?
- I think it is very cumbersome that the annotator has to click save every time they wish to change some annotation, and if you dont click save and move to a new annotation box, it will reset the changes. If I had to change this, I would remove the save button from every annotation and include one for the segment. Once the annotator is happy with the modification he can click save and it will submit and update in the server.

**Comment by yue-su** [comment]
> @yue-su, have you had a chance to look at these points?? The second is especially important.
> 
> Do you think the issue is what Bruno suggested (that all segments are being fetched at once)? I know that adding the foreign key (#7) might have alleviated the problem, but one user can still have created thousands of segments.


Yes, I am working on some improvements regarding the initial data loading after the user logged in. loading thousands of segments should not be a big issue as we have a much larger number of file items(19k) on the sfu vm. The frontend can handle 10k+ items in terms of segments, annotations, etc. 

Let me finish the current branch with the improvements and then we can do another round of testing.
